sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1055,6 +1055,26 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     Exists(plan(ctx.query))
   }
 
+  private def getPredicate(
+      operator: ComparisonOperatorContext): (Expression, Expression) => Expression = {
+    operator.getChild(0).asInstanceOf[TerminalNode].getSymbol.getType match {
+      case SqlBaseParser.EQ =>
+        (left, right) => EqualTo(left, right)
```

Comment:
When I tried to add unit tests by using `assertEqual(sqlCommand, expression)` like 
```scala
assertEqual(
      "a = any (select b from c)",
      AnySubquery(Seq('a), ListQuery(table("c").select('b)), EqualTo))
```
It failed because the '=' operator is parsed to a lambda expression defined in AstBuilder, which is not equal to the function `EqualTo`.
Although I can use `compareExpressions(defaultParser.parseExpression(...),...)` to test it, I think change `(left, right) => xxx(left, right)` to `xxx` here would be better. What do you think? @maropu

Reply:
In that case, how do you handle `Not` in this function?

Reply:
like this:
```scala
private def getPredicate(
      operator: ComparisonOperatorContext): (Expression, Expression) => Expression = {
    operator.getChild(0).asInstanceOf[TerminalNode].getSymbol.getType match {
      case SqlBaseParser.EQ =>
        EqualTo
      case SqlBaseParser.NSEQ =>
        EqualNullSafe
      case SqlBaseParser.NEQ | SqlBaseParser.NEQJ =>
        (left, right) => Not(EqualTo(left, right))
      .....
     }
}
```
Although this is not 'good-looking', it works.

Reply:
IMHO better to handle this case in the test code side...

Reply:
ok, I'll try it.